src/main/java/cn/nukkit/command/data/CommandParameter.java

```diff
@@ -64,6 +64,20 @@ public CommandParameter(String name, boolean optional, String[] enumValues) {
         this.enumData = new CommandEnum(name + "Enums", Arrays.asList(enumValues));
     }
 
+    public CommandParameter(String name, boolean optional, CommandEnum enumData) {
+        this.name = name;
+        this.type = CommandParamType.RAWTEXT;
+        this.optional = optional;
+        this.enumData = enumData;
+    }
+
+    public CommandParameter(boolean optional, String name, String postFix) {
```

Comment:
Why is the name not the first parameter?

Reply:
The constructor `CommandParameter(String, String, boolean)` is already defined. I think it's time to remove the deprecated constructor

Reply:
I think all of the constructors are confusing to anyone trying to use this. Instead it would be a lot easier if we defined a private constructor that has all the fields defined in it then create static factory methods like `CommandParameter.newEnum(String name, boolean optional, String... values)`, `CommandParameter.newType(String name, boolean optional, CommandParamType type)`
Then deprecate all the old constructors.